src/nbl/asset/interchange/CGLTFLoader.cpp

```diff
@@ -171,40 +201,52 @@ namespace nbl
+						if (!image_view_bundle.getContents().empty())
+							cpuImageView = core::smart_refctd_ptr_static_cast<ICPUImageView>(image_view_bundle.getContents().begin()[0]);
+						else
 						{
-							case IAsset::ET_IMAGE:
-							{
-								ICPUImageView::SCreationParams viewParams;
-								viewParams.flags = static_cast<ICPUImageView::E_CREATE_FLAGS>(0u);
-								viewParams.image = core::smart_refctd_ptr_static_cast<asset::ICPUImage>(cpuAsset);
-								viewParams.format = viewParams.image->getCreationParameters().format;
-								viewParams.viewType = IImageView<ICPUImage>::ET_2D;
-								viewParams.subresourceRange.baseArrayLayer = 0u;
-								viewParams.subresourceRange.layerCount = 1u;
-								viewParams.subresourceRange.baseMipLevel = 0u;
-								viewParams.subresourceRange.levelCount = 1u;
-
-								cpuImageView = ICPUImageView::create(std::move(viewParams)); // TODO: cache this image view
-							} break;
-
-							case IAsset::ET_IMAGE_VIEW:
-							{
-								cpuImageView = core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(cpuAsset);
-							} break;
+							auto image_bundle = assetManager->getAsset(glTFImage.uri.value(), context.loadContext.params);
+							if (image_bundle.getContents().empty())
+								return {};
+
+							auto cpuAsset = image_bundle.getContents().begin()[0];
 
-							default:
+							switch (cpuAsset->getAssetType())
 							{
-								context.loadContext.params.logger.log("GLTF: EXPECTED IMAGE ASSET TYPE!", system::ILogger::ELL_WARNING);
-								return {};
+								case IAsset::ET_IMAGE:
+								{
+									ICPUImageView::SCreationParams viewParams;
+									viewParams.flags = static_cast<ICPUImageView::E_CREATE_FLAGS>(0u);
+									viewParams.image = core::smart_refctd_ptr_static_cast<asset::ICPUImage>(cpuAsset);
+									viewParams.format = viewParams.image->getCreationParameters().format;
+									viewParams.viewType = IImageView<ICPUImage>::ET_2D;
+									viewParams.subresourceRange.baseArrayLayer = 0u;
+									viewParams.subresourceRange.layerCount = 1u;
+									viewParams.subresourceRange.baseMipLevel = 0u;
+									viewParams.subresourceRange.levelCount = 1u;
+
+									cpuImageView = ICPUImageView::create(std::move(viewParams)); // TODO: cache this image view
+								} break;
+
+								case IAsset::ET_IMAGE_VIEW:
+								{
+									cpuImageView = core::smart_refctd_ptr_static_cast<asset::ICPUImageView>(cpuAsset);
+								} break;
+
+								default:
+								{
+									context.loadContext.params.logger.log("GLTF: EXPECTED IMAGE ASSET TYPE!", system::ILogger::ELL_WARNING);
+									return {};
+								}
 							}
-						}
+
+							SAssetBundle samplerBundle = SAssetBundle(nullptr, { core::smart_refctd_ptr(cpuImageView) });
+							_override->insertAssetIntoCache(samplerBundle, cpuImageViewCacheKey, context.loadContext, _hierarchyLevel);
+						}	
```

Comment:
turn all of this into a common utility that both glTF and MTL loader can use (they both need to cache on-demand image views)

Reply:
ofc

Reply:
TODO (give issue number here)

Reply:
https://github.com/Devsh-Graphics-Programming/Nabla/issues/270